convert/harness/downgrader/downgrade.go

```diff
@@ -315,13 +317,15 @@ func (d *Downgrader) countGitConnectors(steps []*v1.Step) int {
 	return gitConnectorCount
 }
 
-func (d *Downgrader) extractGitConnectorConfig(steps []*v1.Step) *v0.GitConnector {
+func (d *Downgrader) extractGitConnectorConfig(steps []*v1.Step) (*v0.GitConnector, []*v1.Step) {
 	for stepIdx, step := range steps {
 		if step.Type == "group" {
 			spec := step.Spec.(*v1.StepGroup)
-			connector := d.extractGitConnectorConfig(spec.Steps)
+			connector, newSteps := d.extractGitConnectorConfig(spec.Steps)
+			spec.Steps = newSteps
 			if connector != nil {
-				return connector
+				steps = slices.Delete(steps, stepIdx, stepIdx+1)
```

Comment:
Can you explain the issue this change is fixing? As far as I can tell, the slices.Delete on line 361 should make this one unnecessary.

Reply:
Slices are passed by value. Deleting creates a null step